src/interfaces.d.ts

```diff
@@ -25,6 +25,12 @@ export interface ChildNodeFunction {
 	(this: Widget<WidgetState, WidgetProperties>): DNode[];
 }
 
+export interface PropertiesChangeEvent<T> extends EventTypedObject<'properties:changed'> {
```

Comment:
agreed, looks like this would be similar to the `invalidate` in terms of typing on the `WidgetOverrides`